app/src/main/java/com/example/sharingang/MainActivity.kt

```diff
@@ -80,14 +80,17 @@ class MainActivity : AppCompatActivity(R.layout.activity_main) {
 
     fun onRandomItem(item: MenuItem) {
         lifecycleScope.launch(Dispatchers.IO) {
-            val random = itemRepository.getAll().random()
-            lifecycleScope.launch(Dispatchers.Main) {
-                val drawerLayout = findViewById<DrawerLayout>(R.id.drawer_layout)
-                drawerLayout.close()
-                getNavController().navigate(
-                    ItemsListFragmentDirections
-                        .actionItemsListFragmentToDetailedItemFragment(random)
-                )
+            val allItems = itemRepository.getAll()
+            if (allItems.isNotEmpty()) {
+                val random = allItems.random()
+                lifecycleScope.launch(Dispatchers.Main) {
+                    val drawerLayout = findViewById<DrawerLayout>(R.id.drawer_layout)
+                    drawerLayout.close()
+                    getNavController().navigate(
+                        ItemsListFragmentDirections
+                            .actionItemsListFragmentToDetailedItemFragment(random)
+                    )
+                }
```

Comment:
```suggestion
            val allItems = itemRepository.getAll()
            if (allItems.isEmpty()) {
              return
            }
            val random = allItems.random()
            lifecycleScope.launch(Dispatchers.Main) {
              val drawerLayout = findViewById<DrawerLayout>(R.id.drawer_layout)
              drawerLayout.close()
              getNavController().navigate(
                ItemsListFragmentDirections
                  .actionItemsListFragmentToDetailedItemFragment(random)
              )
            }
```

Let's just use an early return here instead

Reply:
Not allowed apparently